Tests/AsyncHTTPClientTests/HTTPConnectionPool+HTTP2StateMachineTests.swift

```diff
@@ -670,6 +673,89 @@ class HTTPConnectionPool_HTTP2StateMachineTests: XCTestCase {
         XCTAssertTrue(queuer.isEmpty)
     }
 
+    func testMigrationFromHTTP1ToHTTP2WhileShuttingDown() {
```

Comment:
I think this test case is a little to complex... let's just focus on the thing that we actually observed.

1. run request -> opens new connection (expect h1)
2. shutdown pool -> queued request shall be cancelled
3. connection creation is done as h2 -> connection shall be closed immediately and shutdown is done.

then we should also have this vica versa.

1. run request -> opens new connection
2. connection creation is done as h2 -> schedule request on h2 connection
3. connection is done with the single request
4. idle timeout closes h2 connection
5. new request comes in. we create a new connection (expect h2)
6. shutdown pool -> cancel queued request
7. connection creation is done as h1 -> connection shall be closed immediately and shutdown is done.